app/services/form_duplicator.rb

```diff
@@ -10,20 +10,24 @@ def initialize(form)
   end
 
   def duplicate
-    @form.form_elements.map(&:dup).
-      map do |clone|
-        clone.form = new_form
-        clone
-      end.each(&:save)
+    @form.form_elements.each do |element|
+      element = element.dup
+      element.form = new_form
+      element.save
+    end
 
-    new_form.update master: false
     new_form
   end
 
   private
 
   def new_form
-    @new_form ||= @form.dup
+    return @new_form || (
```

Comment:
I think you're going a little overboard with this use of the `||` operator... why not just write
```
return @new_form if @new_form
@new_form = @form.dup
@new_form.master = false
@new_form.save
@new_form
```